crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -229,6 +230,45 @@ where
     }
 }
 
+/// Returns the current system time as Unix timestamp in seconds.
+fn current_time() -> u32 {
+    SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap_or_default()
+        .as_secs() as u32
+}
+
+/// Returns the TTL to set in RRSIG RR and the given Record.
+///
+/// ```text
+/// RFC 4035             DNSSEC Protocol Modifications            March 2005
+///
+/// If the resolver accepts the RRset as authentic, the validator MUST
+/// set the TTL of the RRSIG RR and each RR in the authenticated RRset to
+/// a value no greater than the minimum of:
+///
+///   o  the RRset's TTL as received in the response;
+///
+///   o  the RRSIG RR's TTL as received in the response;
+///
+///   o  the value in the RRSIG RR's Original TTL field; and
+///
+///   o  the difference of the RRSIG RR's Signature Expiration time and the
+///      current time.
+/// ```
+///
+/// See RFC 4035, section 5.3.3: https://datatracker.ietf.org/doc/html/rfc4035#section-5.3.3
+///
+/// **Note** `bind` will substract 3 seconds from the final value to ensure the TTL is slightly
+///          below the difference between signature expiration and current time.
+///
+fn min_ttl(rrsig: &RRSIG, record: &Record, current_time: u32) -> u32 {
```

Comment:
I think `min_ttl` doesn't do a great job of conveying the intention here. Maybe `authenticated_ttl()`? Does it make sense to attach this to `RRSIG` (or `Record`) as a method?

Reply:
Renamed to `authenticated_ttl`.